src/backend/distributed/executor/multi_utility.c

```diff
@@ -630,10 +630,22 @@ ProcessIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand
 
 		if (isDistributedRelation)
 		{
+			Oid namespaceId = InvalidOid;
+			Oid indexRelationId = InvalidOid;
 			ErrorIfUnsupportedIndexStmt(createIndexStatement);
 
-			/* if it is supported, go ahead and execute the command */
-			ExecuteDistributedDDLCommand(relationId, createIndexCommand, isTopLevel);
+			/*
+			 * If the command is supported, and the index does not exist,
+			 * go ahead and execute the command.
+			 */
+			namespaceId = get_namespace_oid(createIndexStatement->relation->schemaname,
```

Comment:
Actually, now that I've looked at this a little longer, I dislike how we duplicate the `get_namespace`/`get_relname` calls between here and `ErrorIfUnsupportedIndexStmt`, which is only called by this function in the first place.

It could be argued that `ErrorIfUnsupportedIndexStmt` shouldn't even be checking for duplicates in the first place, as duplicate indexes aren't actually about whether or not we support a given statement, but about the state of the database system when the `CREATE INDEX` runs.

With that in mind, I think this would be easier to reason about if the "what to do about duplicates" logic were all in one place, i.e. remove the duplicate-checking code from `ErrorIfUnsupportedIndexStmt` and put it in the `if` statement, like so…

```c
char *indexName = createIndexStatement->idxname;
Oid namespaceId = InvalidOid;
Oid indexRelationId = InvalidOid; 

ErrorIfUnsupportedIndexStmt(createIndexStatement);

namespaceId = get_namespace_oid(createIndexStatement->relation->schemaname,
								false);
indexRelationId = get_relname_relid(indexName, namespaceId);

/* If the index name is in use but no `IF EXISTS` option was used, error out */
if (OidIsValid(indexRelationId) && !createIndexStatement->if_not_exists)
{
	ereport(ERROR, (errcode(ERRCODE_DUPLICATE_TABLE),
			errmsg("relation \"%s\" already exists", indexName)));
}

/* Otherwise, it's safe to send the DDL command to worker nodes */
ExecuteDistributedDDLCommand(relationId, createIndexCommand, isTopLevel);
```